File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/Models/Concrete/MacButton.cs
using AbstractFactoryDesignPattern.Example1.Models.Abstract;

namespace AbstractFactoryDesignPattern.Example1.Models.Concrete;

public class MacButton : Button
{
	public override void Render()
	{
		Console.WriteLine("Rendering Mac Button");
	}
	public override void OnClick()
	{
		Console.WriteLine("Mac Button Clicked");
	}
}

File: java/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/main/java/creational/abstractfactory/example1/models/concretes/MacButton.java
package creational.abstractfactory.example1.models.concretes;

import creational.abstractfactory.example1.models.abstracts.Button;

public class MacButton extends Button {
    public void render()   { System.out.println("Rendering Mac Button"); }
    public void onClick()  { System.out.println("Mac Button Clicked"); }
}
